### PR TITLE
[Polygon] Bor: PIP-30: increased max code size limit to 32KB

### PR DESCRIPTION
**Pull request by pratikspatil024**
[Polygon] Bor: [PIP-30: increased max code size limit to 32KB](https://github.com/maticnetwork/Polygon-Improvement-Proposals/blob/main/PIPs/PIP-30.md)

### CODE REVIEW

**Review by ssandeep — APPROVED** [review]

File: core/vm/evm.go [hunk 1]
```diff
@@ -426,11 +426,17 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gasRemainin
...
 
 	// EIP-170: Contract code size limit
-	if err == nil && evm.chainRules.IsSpuriousDragon && len(ret) > params.MaxCodeSize {
+	if err == nil && evm.chainRules.IsSpuriousDragon {
 		// Gnosis Chain prior to Shanghai didn't have EIP-170 enabled,
 		// but EIP-3860 (part of Shanghai) requires EIP-170.
 		if !evm.chainRules.IsAura || evm.config.HasEip3860(evm.chainRules) {
-			err = ErrMaxCodeSizeExceeded
+			if evm.chainConfig.Bor != nil && evm.chainConfig.Bor.IsAhmedabad(evm.Context.BlockNumber) {
+				if len(ret) > params.MaxCodeSizePostAhmedabad {
```
Comment (taratorio) [inline core/vm/evm.go:434]: there's a lot of unnecessary deep nesting that can be simplified if we set a `maxCodeSize` variable before the outmost if.

e.g.

```
var maxCodeSize int
if evm.chainConfig.Bor != nil && evm.chainConfig.Bor.IsAhmedabad(evm.Context.BlockNumber) {
    maxCodeSize = params.MaxCodeSizePostAhmedabad
} else {
    maxCodeSize = params.MaxCodeSizePostAhmedabad
}

if err == nil && evm.chainRules.IsSpuriousDragon && len(ret) > maxCodeSize {
    // same as before
}
```

**Comment by taratorio** [comment]
@pratikspatil024 you will need to open a 2nd PR to `release/2.60` branch if you want this included in Erigon 2 (you most likely do). The `main` branch is Erigon 3, so we need to update both main and release/2.60.

File: core/vm/evm.go (hunk 1 above)
Comment (taratorio) [inline core/vm/evm.go:434]: this introduces deep nesting that can be avoided if we extract this into a `func (evm *EVM) maxCodeSize()` function
e.g.

```
func (evm *EVM) maxCodeSize() int { 
    if evm.chainConfig.Bor != nil && evm.chainConfig.Bor.IsAhmedabad(evm.Context.BlockNumber) {
        return params.MaxCodeSizePostAhmedabad
    }

    return params.MaxCodeSize
}
```

then the if remains unchanged:
```
if err == nil && evm.chainRules.IsSpuriousDragon && len(ret) > evm.maxCodeSize() {
    // same as before
}
```

Comment (pratikspatil024) [reply core/vm/evm.go]: [Done](https://github.com/erigontech/erigon/pull/11792/commits/d29448ee5046fefa2886f6c63d0f9cad831e14f6), thank you!

**Review by taratorio — APPROVED** [review]

**Comment by pratikspatil024** [comment]
> @pratikspatil024 you will need to open a 2nd PR to release/2.60 branch if you want this included in Erigon 2 (you most likely do). The main branch is Erigon 3, so we need to update both main and release/2.60.

Thanks, @taratorio - created [PR#11811](https://github.com/erigontech/erigon/pull/11811) for Erigon-2.

**Pull request merged**